x-pack/plugins/infra/public/common/visualizations/lens/utils.ts

```diff
@@ -101,3 +104,24 @@ export const getXYVisualizationState = (
   hideEndzones: true,
   ...custom,
 });
+
+export const getDefaultReferences = (
+  dataView: DataView,
+  dataLayerId: string
+): SavedObjectReference[] => {
+  return [
+    {
+      type: 'index-pattern',
+      id: dataView.id ?? DEFAULT_AD_HOC_DATA_VIEW_ID,
+      name: `indexpattern-datasource-layer-${dataLayerId}`,
+    },
+  ];
+};
+
+export const getAdhocDataView = (dataView: DataView): Record<string, DataViewSpec> => {
+  return {
+    [dataView.id ?? DEFAULT_AD_HOC_DATA_VIEW_ID]: {
```

Comment:
in what scenario would I not have the dataView.id existing when this function is called?  And in such a case is `adhoc1` unique enough?

Reply:
It is more related to the fact that the `id` is optional in the type definition. But yeah, `adhoc1` is a weak id. I'll give try to make it more unique